security/pkg/nodeagent/caclient/providers/keyfactor/client_test.go

```diff
@@ -0,0 +1,383 @@
+// Copyright 2020 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package caclient
+
+import (
+	"fmt"
+	"os"
+	"testing"
+	"time"
+
+	assert "github.com/stretchr/testify/assert"
+	"golang.org/x/net/context"
+
+	mockKeyfactor "istio.io/istio/security/pkg/nodeagent/caclient/providers/keyfactor/mock"
+	nodeagentutil "istio.io/istio/security/pkg/nodeagent/util"
+	pkiutil "istio.io/istio/security/pkg/pki/util"
+)
+
+var (
+	rootCertPEM = []byte(`-----BEGIN CERTIFICATE-----
```

Comment:
Please add a comment about how these are generated. We have had issues in the past of expired certs in tests breaking. Also make sure the expiration is 10years+